components/formats-bsd/src/loci/formats/out/TiffWriter.java

```diff
@@ -214,19 +226,53 @@ public void saveBytes(int no, byte[] buf, IFD ifd, int x, int y, int w, int h)
     int type = FormatTools.pixelTypeFromString(
         retrieve.getPixelsType(series).toString());
     int index = no;
-    // This operation is synchronized
-    synchronized (this) {
-      // This operation is synchronized against the TIFF saver.
-      synchronized (tiffSaver) {
-        index = prepareToWriteImage(no, buf, ifd, x, y, w, h);
-        if (index == -1) {
-          return;
+    int imageWidth = retrieve.getPixelsSizeX(series).getValue().intValue();
+    int imageHeight = retrieve.getPixelsSizeY(series).getValue().intValue();
+    if (tileSizeX < imageWidth || tileSizeY < imageHeight) {
+      ifd.put(new Integer(IFD.TILE_WIDTH), new Long(tileSizeX));
+      ifd.put(new Integer(IFD.TILE_LENGTH), new Long(tileSizeY));
+    }
+    if (tileSizeX < w || tileSizeY < h) {
+      int nXTiles = w / tileSizeX;
+      int nYTiles = h / tileSizeY;
+   
+      for (int yTileIndex=0; yTileIndex<nYTiles; yTileIndex++) {
+        for (int xTileIndex=0; xTileIndex<nXTiles; xTileIndex++) {
+          byte [] tileBuf = getTile(buf, xTileIndex, yTileIndex, x, y, w, h);
+          int tileX = x + (xTileIndex * tileSizeX);
```

Comment:
Are these `tileX` and `tileY` calculations correct?  What happens when x and y are not aligned to tile boundaries?